core/src/main/scala/org/apache/spark/network/netty/SparkTransportConf.scala

```diff
@@ -31,21 +31,27 @@ object SparkTransportConf {
 
   /**
    * Utility for creating a [[TransportConf]] from a [[SparkConf]].
-   * @param _conf the [[SparkConf]]
-   * @param module the module name
+   * @param _conf         the [[SparkConf]]
+   * @param module         the module name
    * @param numUsableCores if nonzero, this will restrict the server and client threads to only
    *                       use the given number of cores, rather than all of the machine's cores.
    *                       This restriction will only occur if these properties are not already set.
+   * @param role           optional role, could be driver, executor, worker and master. Default is
+   *                      [[None]], means no role specific configurations.
    */
-  def fromSparkConf(_conf: SparkConf, module: String, numUsableCores: Int = 0): TransportConf = {
+  def fromSparkConf(
+      _conf: SparkConf,
+      module: String,
+      numUsableCores: Int = 0,
+      role: Option[String] = None): TransportConf = {
     val conf = _conf.clone
-
-    // Specify thread configuration based on our JVM's allocation of cores (rather than necessarily
-    // assuming we have all the machine's cores).
-    // NB: Only set if serverThreads/clientThreads not already set.
+    val (serverThreads, clientThreads) = getThreadsConfByRole(conf, module, role)
```

Comment:
I think we can simplify this like something below and remove two added functions.
```
val (serverThreads, clientThreads) = role match {
  case Some(r) => (conf.getInt(s"spark.$r.$module.io.serverThreads", numThreads),
    conf.getInt(s"spark.$r.$module.io.clientThreads", numThreads))
  case None => (numThreads, numThreads)
```

Reply:
re-structured code as discussed. thanks.